kernel/src/nr.rs

```diff
@@ -202,6 +223,25 @@ impl Dispatch for KernelNode {
                 }
             }
             Op::SchedAllocateCore(_pid, _affinity, _gtid, _entry_point) => unimplemented!(),
+            Op::SchedReleaseCore(pid, _affinity, gtid) => {
+                #[cfg(not(feature = "rackscale"))]
+                assert!(gtid < crate::arch::MAX_CORES, "Invalid gtid");
```

Comment:
is it `< MAX_XORES*MAX_MACHINES` in the rackscale case?

Reply:
Core naming in rackscale isn't sequential so that's not quite it but I could add a check for the theoretical max gtid value using MAX_MACHINES and MAX_CORES_PER_MACHINE.

However, I'm not sure it's needed - this check doesn't take into account actual architecture, so in any deployment smaller than the max some invalid cores would pass it. I've been assuming the check exists mostly as a debug measure?